src/redux/books/books.js

```diff
@@ -0,0 +1,20 @@
+import { createSlice } from '@reduxjs/toolkit';
+
+const initialState = [];
+
+export const bookSlice = createSlice({
+  name: 'book',
+  initialState,
+  reducers: {
+    addBook: (state, action) => {
+      state.push(action.payload);
+    },
+    removeBook: (state, action) => {
+      state.splice(action.payload);
+    },
+  },
+});
+
+export const { addBook, removeBook } = bookSlice.actions;
+
+export default bookSlice.reducer;
```

Comment:
- [ ] Good job so far but from the project requirement you should follow  [ducks pattern](https://github.com/erikras/ducks-modular-redux)...` const CREATE = 'my-app/widgets/CREATE';` so you should have something like this `const ADD_BOOK = bookstore/books/ADD_BOOK ` and  `const REMOVE_BOOK = bookstore/books/REMOVE_BOOK`. Please fix it to make your code works as expected :smile: 

#### Find an example below :arrow_down: 

```
const ADD_BOOK = bookstore/books/ADD_BOOK ;
const REM_BOOK = bookstore/books/REM_BOOK;

// Reducer Section
export default function reducer(state = [ ], action = {}) {
  switch (action.type) {
    case ADD_BOOK:
      return [
        ...state,
        action.payload,
      ];
    case REM_BOOK:
      return [
        state.filter((item) => item !== action.payload),
      ];
    default:
      return state;
  }
}

// Action Creators Section
export function addBook(obj) {
  return { type: ADD_BOOK, payload: obj };
}

export function removeBook(obj) {
  return { type: ADD_BOOK, payload: obj };
}

```

Reply:
Hi @codecaiine , 
Thank you for your review but I have a few concerns.
I thought the dunk patter is just a way of arranging directories and files. Because i noticed it was stated in the requirements to use redux toolkit which made me write my code in this manner. Please can you explain further?

Reply:
Hi I understand what you mean but Please not that following the Duck patern is required for this project.  I strongly recommend you to check the different examples I shared and trying to follow the duck patern .😊